File: JAVA/S6_LINKEDLIST/L1_1DLINKEDLIST/_2_insert.java



class Node{
    int data;
    Node next;
    Node(int data){
        this.data=data;
        this.next=null;
    }
}
class LinkedList{
    Node head;
    void insertbeg(int data){
        Node newnode=new Node(data);
        newnode.next=head;
        head=newnode;
    }
    void insertlast(int data){
        Node newnode=new Node(data);
        Node temp=head;
        while(temp.next!=null){
            temp=temp.next;
        }
        temp.next=newnode;
    }
    void insertpos(int data,int pos){
        Node newnode=new Node(data);
        if(pos==1){
            newnode.next=head;
            head=newnode;
            return;
        }
        int count=1;
        Node temp=head;
        while(temp!=null && count<pos-1){
            temp=temp.next;
            count++;
        }
       if(temp==null)return;
       newnode.next=temp.next;
       temp.next=newnode;


    }
    void display(){
        Node temp=head;
        while(temp!=null){
            System.out.print(temp.data+" ");
            temp=temp.next;
        }

    }
}
public class _2_insert{
    public static void main(String[] args){
        Node head=null;
        LinkedList list=new LinkedList();
        list.insertbeg(23);
        list.insertbeg(45);
        list.insertpos(65,1);
        list.display();

    }
}

File: python/S6_LINKEDLIST/L1_1DLINKEDLIST/2_insert.py
class Node:
    def __init__(self,data):
        self.next=None
        self.data=data
class List:
    def __init__(self):
        self.head=None
    def insertbeg(self,data):
        newnode=Node(data)
        newnode.next=self.head
        self.head=newnode
    def insertlast(self,data):
        newnode=Node(data)
        temp=self.head
        while temp.next is not None:
            temp=temp.next
        temp.next=newnode
    def display(self):
        temp=self.head
        while temp is not None:
            print(temp.data,end=" ")
            temp=temp.next
if __name__=="__main__":
    l=List()
    l.insertbeg(34)
    l.insertbeg(23)
    l.insertlast(78)
    l.display()


        